### PR TITLE
A list of small improvements

### PR DESCRIPTION
**Pull request by fcollonval**
- Fix dep wrongly specified as dev deps
- Restore CSS scoping
- Reduce diff due to format per Vidar comment
- Restore `getMergedValue`
- Improve integration tests config
  - Reduce the test matrix (no point to test multiple Python)
  - Test the package installed from wheel not dev
  - Remove unused ref snapshots
- Add integration test for checking resolved conflict download
- Symlink example to ui test to avoid content duplication
- Instantiate editor registries only once (propagate them from top to bottom to reuse tokens in lab)
- Improve API by switching to object instead of args array in all places changes were needed to the API
- Don't clean the favicon when running `npm run clean`
- Fix lumino and lab styles not being deduplicated in the webapp (the app is much more reactive and FF stops crashing when using the dev tools)
- Use only prebuilt lab extension
- Modernize Jest config

### CODE REVIEW

**Comment by fcollonval** [comment]
cc @krassowski for awareness

**Comment by krassowski** [comment]
If that helps I have the actions enabled on my `master` branch, so the new snapshots can be grabbed from https://github.com/krassowski/nbdime/pull/5 (or by opening a PR and triggering bot via comment to get them directly).

**Comment by fcollonval** [comment]
Thanks a lot Mike, I'll take the new snapshots from your branch.

File: ui-tests/tests/nbdime-merge-test1.spec.ts [hunk 1]
```diff
@@ -13,26 +13,40 @@ test.beforeEach(  async ({ page }) => {
...
-  await page.getByText('⚠').click();
-  expect(await page.locator('#main').screenshot()).toMatchSnapshot();
-})
-
-test('choose right version for conflict',  async ({ page }) => {
-  await page.locator('div:nth-child(3) > .cm-editor > .cm-scroller > .cm-gutters > div:nth-child(2) > div:nth-child(2) > .jp-Merge-gutter-picker').click();
-  await page.getByText('⚠').click();
-  expect(await page.locator('#main').screenshot()).toMatchSnapshot();
-})
+  test('open an example and take a snapshot', async ({ page }) => {
```
Comment (fcollonval) [inline ui-tests/tests/nbdime-merge-test1.spec.ts]:
```suggestion
  test('take a snapshot at opening', async ({ page }) => {
```

**Pull request merged**